src/main/scala/edu/berkeley/cs/rise/opaque/OpaqueServer.scala

```diff
@@ -0,0 +1,144 @@
+package edu.berkeley.cs.rise.opaque
+
+import java.util.logging.Logger
+
+import com.google.protobuf.ByteString
+
+import io.grpc.{Server, ServerBuilder}
+
+import ra.Ra
+import ra.GreeterGrpc
+
+import scala.concurrent.{ExecutionContext, Future}
+import org.apache.spark.sql.SparkSession
+
+import implicits._
+
+/**
+ * [[https://github.com/grpc/grpc-java/blob/v0.15.0/examples/src/main/java/io/grpc/examples/helloworld/HelloWorldServer.java]]
+ */
+object OpaqueServer {
```

Comment:
If my understanding is correct, this is the gRPC server listening at the driver, and the gRPC server essentially relays gRPC requests from the client and forwards to the driver. If so, can we rename this to something like `OpaqueRPCListener"